### PR TITLE
Implement OTLP Summary support

### LINKED ISSUE #149: Prometheus Summary conversion is incorrect
The OTLP output for a Prometheus Summary is not properly combining the _sum, _count, and assorted quantiles into a single data point.

### PR DESCRIPTION
**Pull request by carlosalberto**
Follow up after reverting #193 

* Create a single Summary point instead of sending multiple values to be normalized by the backend.

New:

* Do not try to call `getMetricName()` for Summary points without `_sum`/`_count` prefix (specially bad if mixed with custom metric prefix, hence added a small test case for it too).
* Remove any `quantile` label from the final label set, as Histograms do.
* Double check the type when building Summary points - there's an edge case where the base name, labels and timestamp are all the same, but the type is not (and we can't count on `suffix` to differentiate the value, as Histogram _can_).

Fixes #149.

### CODE REVIEW

File: retrieval/transform.go [hunk 1]
```diff
@@ -272,6 +274,110 @@ func (d *distribution) Swap(i, j int) {
...
+				Value: v,
+			})
+		default:
+			break Loop
+		}
+		consumed++
+	}
+	// Don't emit a sample if we explicitly skip it or no reset timestamp was set because the
+	// count series was missing.
+	if skip || resetTimestamp == 0 {
```
Comment (jmacd) [inline retrieval/transform.go]: nit: skip is a dead variable. will you remove it here (and in `buildHistogram()` where it is equally dead)?

**Review by jmacd — APPROVED** [review]

File: retrieval/transform.go (hunk 1 above)
Comment (carlosalberto) [reply retrieval/transform.go]: Oh yesss, had totally forgotten about that one.

Comment (carlosalberto) [reply retrieval/transform.go]: Oh yesss, had totally forgotten about that one.
